riscv/mmu.cc

```diff
@@ -560,6 +570,35 @@ reg_t mmu_t::s2xlate(reg_t gva, reg_t gpa, access_type type, access_type trap_ty
   }
 }
 
+bool mmu_t::check_svukte_qualified(reg_t addr, reg_t mode, bool forced_virt)
+{
+  state_t* state = proc->get_state();
+
+  if (mode != PRV_U)
+    return true;
+
+  if (proc->extension_enabled('S') && get_field(state->senvcfg->read(), SENVCFG_UKTE)) {
+    if (forced_virt && state->prv == PRV_U) {
+      bool hstatus_hukte = proc->extension_enabled('H') &&
+                           (get_field(state->hstatus->read(), HSTATUS_HUKTE) == 1);
```

Comment:
This would read better as `bool hstatus_hukte = proc->extension_enabled('H') && get_field(state->hstatus->read(), HSTATUS_HUKTE);`